ipaserver/install/dogtaginstance.py

```diff
@@ -275,6 +279,64 @@ def uninstall(self):
             logger.critical("failed to uninstall %s instance %s",
                             self.subsystem, e)
 
+    def secure_ajp_connector(self):
+        """ Update AJP connector to use a password protection  """
+        server_xml = open(paths.PKI_TOMCAT_SERVER_XML, 'r').read()
+
+        # AJP connector is set on port 8009. Use non-greedy search to find it
+        connector = re.search('<Connector port="8009".+?/>', server_xml)
+
+        # no AJP connector means no need to update anything
+        if connector is None:
+            return
+
+        # Detect tomcat version and choose the right option name
+        # pre-9.0.31.0 uses 'requiredSecret'
+        # 9.0.31.0 or later uses 'secret'
+        version = ('pre-9.0.31.0', ' requiredSecret="')
```

Comment:
This version tuple ... is it necessary to use a single variable?

version[0] is only used  once basically, to see what version we have (twice if you count the exception).
version[1] is used a bunch of times.